File: app/src/main/java/uk/co/catedroid/app/ui/LaunchActivity.java
package uk.co.catedroid.app.ui;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import uk.co.catedroid.app.CATeApplication;
import uk.co.catedroid.app.auth.LoginManager;

public class LaunchActivity extends AppCompatActivity {

    @Inject
    SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((CATeApplication) getApplication()).getNetComponent().inject(this);
        LoginManager loginManager = new LoginManager(sharedPreferences);

        Intent intent;

        if (loginManager.hasStoredCredentials()) {
            intent = new Intent(this, DashboardActivity.class);
        } else {
            intent = new Intent(this, LoginActivity.class);
        }

        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/uk/co/catedroid/app/ui/LaunchActivity.kt
package uk.co.catedroid.app.ui

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import javax.inject.Inject

import uk.co.catedroid.app.CATeApplication
import uk.co.catedroid.app.auth.LoginManager

class LaunchActivity : AppCompatActivity() {

    @Inject
    lateinit var sharedPreferences: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        (application as CATeApplication).netComponent.inject(this)
        val loginManager = LoginManager(sharedPreferences)

        val intent: Intent

        intent = if (loginManager.hasStoredCredentials()) {
            Intent(this, MainActivity::class.java)
        } else {
            Intent(this, LoginActivity::class.java)
        }

        startActivity(intent)
        finish()
    }
}
